### PR TITLE
wpc-458-highlight-pending-articles-and-pages

### PR DESCRIPTION
**Pull request by HanaAyman**
This pull request is to highlight the state column in articles and pages with different colors depending on the state.

Screen Shot 2021-08-30 at 6 02 18 PM

### CODE REVIEW

**Review by bishoymelek — APPROVED** [review]

**Review by bishoymelek — APPROVED** [review]

**Review by penk85 — DISMISSED** [review]

File: packages/editor/src/client/routes/articleList.tsx [hunk 1]
```diff
@@ -133,7 +133,7 @@ export function ArticleList() {
           data={articles}
           sortColumn={sortField}
           sortType={sortOrder}
-          rowClassName={rowData => (rowData?.id === highlightedRowId ? 'highlighted-row' : '')}
+          rowClassName={(rowData: ArticleRefFragment) => (rowData?.pending ? 'pending-approval' : '' || rowData?.id === highlightedRowId ? 'highlighted-row' : '')}
```
Comment (roosnic1) [inline packages/editor/src/client/routes/articleList.tsx:136]: @HanaAyman what if the row is pending approval and we want to highlight it? We need to be able to give the rows a basic color and still highlight them

**Review by roosnic1 — CHANGES_REQUESTED** [review]

**Review by bishoymelek — APPROVED** [review]
🎉

File: packages/editor/src/client/routes/articleList.tsx [hunk 2]
```diff
@@ -194,7 +194,22 @@ export function ArticleList() {
...
+                    style={{
+                      textAlign: 'center',
+                      borderRadius: '15px',
+                      backgroundColor: rowData.pending
+                        ? 'rgb(248, 222, 242)'
+                        : rowData.published
+                        ? 'rgb(225, 248, 222)'
+                        : rowData.draft
+                        ? 'rgb(248, 239, 222)'
+                        : 'inheret'
```
Comment (penk85) [inline packages/editor/src/client/routes/articleList.tsx:208]: inherit

File: packages/editor/src/client/routes/articleList.tsx [hunk 3]
```diff
@@ -194,7 +194,22 @@ export function ArticleList() {
...
                 if (rowData.published) states.push(t('articles.overview.published'))
 
-                return <div>{states.join(' / ')}</div>
+                return (
+                  <div
+                    style={{
+                      textAlign: 'center',
+                      borderRadius: '15px',
+                      backgroundColor: rowData.pending
+                        ? 'rgb(248, 222, 242)'
```
Comment (penk85) [inline packages/editor/src/client/routes/articleList.tsx:203]: Can you set the colors in a shared object or enum so it's easier to edit? There's a similar one already in the utilities file to copy or reuse.

**Review by penk85 — CHANGES_REQUESTED** [review]
Looks very nice. I'd advise making the colors reusable though.

File: packages/editor/src/client/routes/articleList.tsx (hunk 3 above)
Comment (bishoymelek) [reply packages/editor/src/client/routes/articleList.tsx]: @penk85 thanks a lot for the note, fixed. Please review.

**Review by penk85 — APPROVED** [review]
:+1:

**Review by camilleory — APPROVED** [review]

**Review by roosnic1 — APPROVED** [review]
👍🏾

**Pull request merged**